src/main.cpp

```diff
@@ -145,6 +145,7 @@ BaseSurface::~BaseSurface() {
 }
 
 void BaseSurface::Run() {
+	this->_peakMixerUpdate(); // ToDo: Maybe update only every 2nd call to save CPU?
```

Comment:
Rather than putting this in BaseSurface, you could just override Run() in NiMidiSurface and call the superclass: `BaseSurface::Run()`

Reply:
yes, that would look cleaner and is also more logical (I don't even know if Mk1 has meters - do you?)